File: core/src/main/java/heroes/journey/initializers/base/actions/LoadOptions.java
package heroes.journey.initializers.base.actions;

import heroes.journey.entities.actions.options.BooleanOptionAction;
import heroes.journey.initializers.InitializerInterface;
import heroes.journey.utils.input.Options;

public class LoadOptions implements InitializerInterface {

    public static BooleanOptionAction debugOption, autoEndTurnOption, backgroundMusic;

    @Override
    public void init() {
        debugOption = new BooleanOptionAction("debug", "Debug", false).register();
        Options.addOption(debugOption);

        autoEndTurnOption = new BooleanOptionAction("auto_end_turn", "Auto End Turn", false).register();
        Options.addOption(autoEndTurnOption);

        backgroundMusic = new BooleanOptionAction("music", "Music", false).register();
        Options.addOption(backgroundMusic);
    }
}

File: core/src/main/java/heroes/journey/initializers/base/actions/LoadOptions.kt
package heroes.journey.initializers.base.actions

import heroes.journey.entities.actions.booleanOptionAction
import heroes.journey.entities.actions.options.BooleanOptionAction
import heroes.journey.initializers.InitializerInterface
import heroes.journey.utils.input.Options

class LoadOptions : InitializerInterface {
    override fun init() {
        debugOption = booleanOptionAction {
            id = "debug"
            name = "Debug"
            isTrue = false
        }.register()
        Options.addOption(debugOption!!)

        backgroundMusic = booleanOptionAction {
            id = "music"
            name = "Music"
            isTrue = false
        }.register()
        Options.addOption(backgroundMusic!!)
    }

    companion object {
        @JvmField
        var debugOption: BooleanOptionAction? = null

        @JvmField
        var backgroundMusic: BooleanOptionAction? = null
    }
}
